### PR TITLE
add forwarding latency statistic for mqtt

### PR DESCRIPTION
**Pull request by hui6075**
Hi,
I have add the "stamped" label for mqtt publish request which can give a measurement for the message forwarding latency between publish time and receive time. Same as xmpp_msg_latency in Jabber module.
As we know, mqtt is mainly used in IOT environment, e.g., use the mobile device to control some house appliance. So the forwarding latency is quite a performance target to evaluate a mqtt broker.
Usage:
make stamped="true" in mqtt.xml, i.e.,
<mqtt type="publish" topic="test_topic" qos="1" retained="true" _stamped="false"_>test_message</mqtt>

### CODE REVIEW

**Comment by tisba** [comment]
Same as with https://github.com/processone/tsung/pull/272, can you review this, @nniclausse?

Also, @hui6075, thanks for your first contribution! Could you please take care of the conflicting files? Thanks!

**Comment by hui6075** [comment]
@tisba Yes, I would like to resolve the conflict.
@nniclausse thanks for the comments!

**Comment by tisba** [comment]
Nice! Could you add some documentation as well, @hui6075?

I was also wondering if this is a MQTT feature, or if it just changing the payload which needs to be taken care of at the MQTT server/broker?

**Comment by hui6075** [comment]
@tisba Thank you, this just add the time stamp ahead of the payload, same with the time stamp feature(described in Tsung 1.7.0 documentation section 6.6.3.1) in Tsung XMPP module. How should I add the documentation?

**Comment by tisba** [comment]
@hui6075 This can very simple IMO. Maybe just like in 6.6.3.1, but for MQTT.

**Comment by tisba** [comment]
@hui6075 could you add this to https://github.com/processone/tsung/blob/develop/docs/conf-sessions.rst#mqtt?

**Comment by hui6075** [comment]
@tisba sorry for late due to some personal issue. conf-sessions.rst was updated, could you give a review?

File: docs/conf-sessions.rst [hunk 1]
```diff
@@ -1035,6 +1035,27 @@ Example with MQTT as a session type:
...
+inconsistencies of timestamps on different Tsung nodes.
+
+Only a fraction of requests will hit the same node they originated from,
+but with request rate high enough this fraction should be sufficient.
+
+``stamped`` is allowed only with ``size`` attribute. ``data`` will cause
+``stamped`` to be ignored. There is a minimal length of the stamp,
+roughly 30 bytes. When ``size`` is greater than stamp length, random
+padding will be added to the stamp. If the stamp length is higher than
+``size``, then only stamp will be used as messagecontent, effectively
```
Comment (tisba) [inline docs/conf-sessions.rst]: typo -> `message content`

Also, maybe we can add a short example with `stamped="true"`? I'm not that familiar with MQTT, but reading this, I'm not sure what the `size` attribute is. `data` I assume is the message content?

**Comment by tisba** [comment]
No worries, thank you for keep this going! :)

File: docs/conf-sessions.rst (hunk 1 above)
Comment (hui6075) [reply docs/conf-sessions.rst]: yes, “messagecontet”  and  ``size`` was copied from line 301 in this file, and ``size`` is the attribute for Jabber. I would remove ``size``.
``stamped`` is just a switch to enable/disable the time stamp feature, and the time stamp feature is independent with other mqtt feature, so I think the original MQTT example is enough.

**Comment by tisba** [comment]
Sorry for taking me so long to take another look :( Thank you for your contribution 🎉

**Comment by hui6075** [comment]
you are welcome:)

**Pull request merged**